Sources_v3/StreamChat/ChatClient_Tests.swift

```diff
@@ -373,58 +374,345 @@ class ChatClient_Tests: StressTestCase {
+            tokenProvider: .development(userId: .unique),
+            workerBuilders: [],
+            eventWorkerBuilders: [],
+            environment: .mock
+        )
+
+        // Change the token provider to return a token for different user.
+        client.tokenProvider = .anonymous
+
+        // Call `reloadUserIfNeeded` and catch the error.
+        var reloadUserIfNeededError: Error?
+        client.reloadUserIfNeeded {
+            reloadUserIfNeededError = $0
+        }
+
+        // Assert `ClientError.ClientIsNotInActiveMode` is propagated.
+        AssertAsync.willBeTrue(reloadUserIfNeededError is ClientError.ClientIsNotInActiveMode)
+    }
+
+    func test_reloadUserIfNeeded_connectsToWebSocket_dependingOnShouldConnectAutomatically() {
+        for shouldConnectAutomatically in [true, false] {
+            // Create a config.
+            var config = inMemoryStorageConfig
+            config.shouldConnectAutomatically = shouldConnectAutomatically
+
+            // Create passive client.
+            let client = ChatClient(
+                config: config,
+                tokenProvider: .development(userId: .unique),
+                workerBuilders: [],
+                eventWorkerBuilders: [],
+                environment: .mock
+            )
+
+            // Remember current connect counter.
+            let oldConnectionCounter = client.mockWebSocketClient.connect_calledCounter
+
+            // Call `reloadUserIfNeeded`.
+            client.reloadUserIfNeeded()
+
+            // Assert `ClientError.ClientIsNotInActiveMode` is propagated.
+            if shouldConnectAutomatically {
+                AssertAsync.willBeEqual(client.mockWebSocketClient.connect_calledCounter, oldConnectionCounter + 1)
+            } else {
+                AssertAsync.staysEqual(client.mockWebSocketClient.connect_calledCounter, oldConnectionCounter)
+            }
+        }
+    }
+
+    func test_reloadUserIfNeeded_changesConnectionStatus_forPassiveClients() throws {
+        // Create current user id.
+        let currentUserId: UserId = .unique
+
+        // Create environment for passive client, build the database and save current user.
+        // We need this step to get the current user from database in `ChatClient.init`
+        // (e.g. restore the current user session)
+        //
+        // This step is required because when there's no user the `currentUserId` is nil
+        // and giving any token for any user will fails since passive clients don't allow
+        // switching the user.
+        var environment = ChatClient.Environment.mock
+        let database = try environment.databaseContainerBuilder(.inMemory, true, false)
+        try database.createCurrentUser(id: currentUserId)
+        environment.databaseContainerBuilder = { _, _, _ in database }
+
+        // Create passive client
+        let passiveClient = ChatClient(
+            config: inactiveInMemoryStorageConfig,
```

Comment:
nit: can we please rename `inactiveInMemoryStorageConfig` -> `passiveInMemoryStorageConfig`